src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/AvoidUninstantiatedInternalClasses.cs

```diff
@@ -302,14 +302,8 @@ private bool HasInstantiatedNestedType(INamedTypeSymbol type, IEnumerable<INamed
                 return true;
             }
 
-            // Ignore type generated for holding top level statements
-            if (type.IsTopLevelStatementsEntryPointType())
-            {
-                return true;
-            }
-
             // The type containing the assembly's entry point is OK.
-            if (ContainsEntryPoint(type, compilation))
```

Comment:
On roslyn side, this is calculated once for the compilation and cached:

https://github.com/dotnet/roslyn/blob/main/src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs#L1676-L1684

@mavasani Do you think this PR can contribute to https://github.com/dotnet/roslyn-analyzers/issues/6301

I'm not sure how much the existing code was expensive. Specifically, we could have been calling `GetMembers` for lots of types. Did the compiler already cached them before invoking the analyzer? or does the analyzer come through this code path:

https://github.com/dotnet/roslyn/blob/1100e56a04e144352af43181df077a13957e6a58/src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs#L1656-L1665

Reply:
Regardless of whether or not this addresses https://github.com/dotnet/roslyn-analyzers/issues/6301, it seems to definitely be more hardened code and should likely also improve performance.